tests/testthat/test-cache.R

```diff
@@ -434,6 +434,8 @@ test_that("it works outside of a package", {
 })
 
 test_that("cache = TRUE workflow works", {
+  skip_if_not_installed("cyclocomp")
```

Comment:
Why not drop it from the test package instead?

Reply:
Sorry, don't follow you. Why would we want to do that?

Reply:
Hmm, I am not seeing why this test should fail absent cyclocomp, can you please record how this test fails without cyclocomp installed?

Reply:
Wow, this wasn't needed at all. 

I am not sure why I had added it; my hunch is it was a different issue that affected this test, and I mistook it to be related to cyclocomp linter removal and added this skip.

Reply:
Hmm, the tests fail on CI:

```r
i Please install the needed cyclocomp package.
Backtrace:
     ▆
  1. ├─lintr::lint_package(pkg, cache = TRUE) at test-cache.R:445:3
  2. │ └─lintr::lint_dir(...) at lintr/R/lint.R:263:3
  3. │   ├─lintr:::flatten_lints(...) at lintr/R/lint.R:185:3
  4. │   │ └─lintr:::flatten_list(x, class = "lint") at lintr/R/utils.R:18:3
  5. │   │   └─lintr (local) assign_item(x) at lintr/R/utils.R:36:3
  6. │   └─base::lapply(...) at lintr/R/lint.R:185:3
  7. │     └─lintr (local) FUN(X[[i]], ...)
  8. │       └─lintr::lint(file, ..., parse_settings = FALSE, exclusions = exclusions) at lintr/R/lint.R:189:7
  9. │         ├─base::withCallingHandlers(...) at lintr/R/lint.R:88:9
 10. │         └─lintr:::get_lints(...) at lintr/R/lint.R:88:9
 11. │           ├─lintr:::flatten_lints(linter_fun(expr)) at lintr/R/lint.R:305:5
 12. │           │ └─lintr:::flatten_list(x, class = "lint") at lintr/R/utils.R:18:3
 13. │           │   └─lintr (local) assign_item(x) at lintr/R/utils.R:36:3
 14. │           └─lintr (local) linter_fun(expr) at lintr/R/lint.R:305:5
 15. │             └─cli::cli_abort(...) at lintr/R/cyclocomp_linter.R:28:7
 16. │               └─rlang::abort(...)
 17. │                 └─rlang:::signal_abort(cnd, .file)
 18. │                   └─base::signalCondition(cnd)
 19. └─lintr (local) `<fn>`(`<rlng_rrr>`)
```

Reply:
Interesting... I see cyclocomp referenced in that package, but only in comments:

```shell
grep -Fr cyclocomp tests/testthat/dummy_packages/package
# tests/testthat/dummy_packages/package/inst/data-raw/default_linter_testcode.R:# cyclocomp
# tests/testthat/dummy_packages/package/data-raw/default_linter_testcode.R:# cyclocomp
# tests/testthat/dummy_packages/package/R/default_linter_testcode.R:# cyclocomp
```

Reply:
Maybe there's a cache issue in CI? I confirm it passes locally:

```
remove.packages("cyclocomp")
# Removing package from ‘/home/michael/R/x86_64-pc-linux-gnu-library/4.4’
# (as ‘lib’ is unspecified)
pkgload::load_all()
# ℹ Loading lintr
testthat::test_file('tests/testthat/test-cache.R')
# ══ Testing test-cache.R ══════════════════════════════════════════════════════════════
# [ FAIL 0 | WARN 0 | SKIP 0 | PASS 62 ] Done!
```

Reply:
Actually, I _do_ reproduce the error locally under `R CMD check`:

```
* checking tests ...
  Running ‘testthat.R’
 ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
    9. │         ├─base::withCallingHandlers(...)
   10. │         └─lintr:::get_lints(...)
   11. │           ├─lintr:::flatten_lints(linter_fun(expr))
   12. │           │ └─lintr:::flatten_list(x, class = "lint")
   13. │           │   └─lintr (local) assign_item(x)
   14. │           └─lintr (local) linter_fun(expr)
   15. │             └─cli::cli_abort(...)
   16. │               └─rlang::abort(...)
   17. │                 └─rlang:::signal_abort(cnd, .file)
   18. │                   └─base::signalCondition(cnd)
   19. └─lintr (local) `<fn>`(`<rlng_rrr>`)
  
  [ FAIL 1 | WARN 0 | SKIP 346 | PASS 5901 ]
  Error: Test failures
  Execution halted
```

Reply:
I don't follow _precisely_ the issue, but the error did ring a bell. I think the wrong config was getting picked up, but I'm not sure which config / not sure why cyclocomp was involved.

This test is better off specifying the right lintr config anyway, and doing so made the tests pass (locally at least), so I'm OK to proceed.

Reply:
Thanks for fixing it!

I am fairly certain that this is also the reason the tests fail [in random order](https://github.com/r-lib/lintr/pull/1938).